dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -883,12 +889,14 @@ TRASH_CanTrashFile(LPCWSTR wszPath)
         return FALSE;
     }
 
-    // Only keep the base path.
+    // Create a wide character buffer and copy the given path
     WCHAR wszRootPathName[MAX_PATH];
-    strcpyW(wszRootPathName, wszPath);
-    PathRemoveFileSpecW(wszRootPathName);
-    PathAddBackslashW(wszRootPathName);
+    strcpyW(wszRootPathName, wszPath); // warning: possible buffer overflow
```

Comment:
You can fix it by using StringCbCopyW instead

Reply:
Thanks, fixed in the next commit